docs/user_docs/concepts/package-and-module.md

```diff
@@ -0,0 +1,154 @@
+
+Within the same module, different packages can be imported from each other through the import statement of relative or absolute path. In the KCL, the relative import will be replaced by absolute import and the corresponding KCL code will be found through the package path.
+
+### Relative Import Path
+
+We can use the operator `.` to realize the relative path import of KCL entry files.
+
+main.k:
+
+```python
+import .model1  # Current directory module
+import ..service  # Parent directory
+import ...root  # Parent of parent directory
+
+s = service.ImageService {}
+m = root.Schema {}
+```
+
+### Absolute Import Path
+
+The semantics of `import a.b.c.d` is
+
+1. If `kcl.mod` not exist, search the path `./a/b/c/d` from the current directory.
+2. If the current directory search fails, search from the root path `ROOT_PATH/a/b/c/d`, else raise a import error.
+
+The definition of the root path `ROOT_PATH` is the directory corresponding to the `kcl.mod` file from the current directory.
+
+Code structure:
+
+```
+. 
+└── root
+    ├── kcl.mod
+    ├── model
+    │   ├── model1.k
+    |   ├── model2.k
+    │   └── main.k
+    ├── service
+    │   └── service1.k
+    └── mixin
+        └── mixin1.k
+```
+
+### Builtin Package
+
+KCL has a collection of builtin packages such as `math`, `regex`, etc. To use a builtin package, import it directly and invoke the functions using its qualified identifier. For instance,
+
+```python
+import regex
+
+image = "nginx:1.14.2"
+is_match = regex.match(image, "^[a-zA-Z]+:\d+\.\d+\.\d+$")
+
+```
+
+The output YAML is
+
+```yaml
+image: nginx:1.14.2
+is_match: true
+```
+
+### Plugin Package
+
+KCL also has a collection of plugin packages such as `hello`, `project_context`, etc. To use a plugin package, import it with a `kcl_plugin.` package path prefix and invoke the functions using its qualified identifier. For instance,
+
+```python
+import kcl_plugin.hello
+
+result = hello.add(1, 1)
+```
+
+The output YAML is
+
+```yaml
+result: 2
+```
+
+### Main Package
+
+In KCL, the composition of the main package is usually determined by the compiler parameters. This is because the KCL schema and constraints can be split across files in the package, or even organized across directories, considering the convenience of writing and maintaining the configuration in isolated blocks.
+
+#### Files belonging to a main package
+
+It is up to the user to decide which configurations and constraints to use using the KCL command line. For example,
+
+```bash
+kcl file1.k file2.k
+```
+
+Thus, the main package contains two KCL files named `file1.k` and `file2.k`.
+
+If KCL is told to load the files for a specific directory, for example:
+
+```bash
+kcl ./path/to/package
```

Comment:
if the `./path/to/package` contains `kcl.yaml` file, will the `kcl.yaml` file be recognized as a settings file, or be ignored?